guides/introduction/negotiation.md

```diff
@@ -115,75 +115,99 @@ Now we create the answer, set it, and send it back to the web browser.
+
+## ICE and candidate exchange
+
+ICE is a protocol used by WebRTC to establish peer-to-peer connection. It works by exchanging something called _ICE candidates_
+between the peers using some kind of separate medium (similar to the offer/answer exchange). These candidates, simplifying a bit, contain IP addreses that other
+peer will try to use to connect to your machine. ICE will try to find a pair of these addresse (one for each peer) and establish a connection.
+
+> #### Why candidates are not in the offer/answer? {: .info}
+> ICE candidates can be included in the offer or the answer, but generally they are not - you send them separately.
+> Gathering a candidate can be anywhere in-between of nearly instantenous, or taking up to a few seconds, depending on the type of the candidate.
+> The PeerConnection will asynchornously produce the "quicker" candidates so you can send them to the other peer and try to establish
+> a connection as quickly as possible. If any of the later candidates happens to be more suitable (or the previous did not succeed), PeerConnection will use it instead.
+
+The PeerConnection will gather these candidates, but it is your responsibility (similarly to offer/answer exchange, again) to send them to the other peer.
+
+In JavaScript:
+
+```js
+// the end of candidates will be signalled by event.candidate === null
+pc.onicecandidate = event => webSocket.send_ice_candidate(JSON.stringify(event.candidate));
+webSocket.onIceCandidate = candidate => pc.addIceCandidate(JSON.parse(candidate));
+```
+
+And in Elixir:
+
+```elixir
+receive do
+  {:ex_webrtc, ^pc, {:ice_candidate, candidate}} ->
+    candidate
+      |> ExWebRTC.ICECandidate.to_json()
+      |> Jason.encode!()
+      |> web_socket_send_ice_candidate()
+
+  {:web_socket, {:ice_candidate, json}} ->
+    candidate =
+      json
+      |> Jason.decode!()
+      |> ExWebRTC.ICECandidate.from_json()
+
+      ExWebRTC.PeerConnection.add_ice_candidate(pc, candidate)
+end
+```
+
+After the candidate exchange, the connection should be eventually established and media will start to flow!
+You can tell it by listening for `{:ex_webrtc, _from, {:connection_state_change, :connected}}` message
+or by handling the `onconnectionstatechange` event on the JavaScript `RTCPeerConnection`.
+
+> #### ICE servers {: .info}
+> Remember when we created the `RTCPeerConnection` object at the beginning of this tutorial? It was configured
+with `iceServers` options:
+>
+> ```js
+> const pc = new RTCPeerConnection({ iceServers: "stun:stun.l.google.com:19302" })
+> ```
+>
+> It is a list of STUN/TURN servers that the PeerConnection will try to use. You can learn more about
+> it in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection) but
+> it boils down to the fact that lack of any STUN servers might cause you trouble connecting with other peers, so make sure
```

Comment:
I think that it will be easier to mention that iceServers are used to discover more ICE candidates when we are behind NAT.

You think about it as: okay ICE generates some candidates but those might not be enough so we provide external servers to discover more of them